packages/maria/src/lib/MariaProvider.ts

```diff
@@ -726,8 +738,10 @@ export class MariaProvider<StoredValue = unknown> extends JoshProvider<StoredVal
     return payload;
   }
 
-  protected fetchVersion() {
-    return this.version;
+  protected async fetchVersion() {
+    const metadata = await this.handler.fetchMetadata();
+
+    return resolveVersion(metadata.version);
```

Comment:
Does this default to resolving the [VI] version?

Reply:
If fetchVersion() is called after the handler is initialized, then I believe it will default to [VI]version[/VI].
It'll fetch the metadata which is ensured in the handler.init() using provider.version which returns [VI]version[/VI].